include/deal.II/lac/schur_complement.h

```diff
@@ -0,0 +1,498 @@
+    const auto S = schur_complement(A_inv,B,C,D);
+    const auto S_prec = PreconditionSelector<...>(D); // D and S operate on same space
+    const auto S_inv = inverse_operator<...>(S,...,prec_S);
+
+    // Solve reduced block system
+    auto rhs = condense_schur_rhs (A_inv,C,f,g); // PackagedOperation that represents the condensed form of g
+    y = S_inv * rhs; // Solve for y
+    x = postprocess_schur_solution (A_inv,B,y,f); // Compute x using resolved solution y
+   @endcode
+ *
+ * In the above example, the preconditioner for \f$ S \f$ was defined as the
+ * preconditioner for \f$ D \f$, which is valid since they operate on the same
+ * space.
+ * However, if \f$ D \f$ and \f$ S \f$ are too dissimilar, then this may lead to
+ * a large number of solver iterations as \f$ \text{prec}(D) \f$ is not a good
+ * approximation for \f$ S^{-1} \f$.
+ *
+ * A better preconditioner would be therefore one that provides a more
+ * representative approximation for \f$ S^{-1} \f$.
+ * This can be achieved by firstly constructing an object that represents an
+ * approximation for \f$ S \f$ wherein expensive operation, namely
+ * \f$ A^{-1} \f$, is approximated.
+ * Thereafter we construct the approximate inverse operator \f$ \tilde{S}^{-1} \f$
+ * which is then used as the preconditioner for computing \f$ S^{-1} \f$.
+ * @code
+    // Construction of approximate inverse of Schur complement
+    const auto A_inv_approx = linear_operator(preconditioner_A);
+    const auto S_approx = schur_complement(A_inv_approx,B,C,D);
+    const auto S_approx_prec = PreconditionSelector<...>(D); // D and S_approx operate on same space
+    const auto S_inv_approx = inverse_operator(S_approx,...,S_approx_prec); // Inner solver: Typically limited to few iterations using IterationNumberControl
+
+    // Construction of exact inverse of Schur complement
+    const auto S = schur_complement(A_inv,B,C,D);
+    const auto S_inv = inverse_operator(S,...,S_inv_approx); // Outer solver
+
+    // Solve reduced block system
+    auto rhs = condense_schur_rhs (lo_A_inv,lo_C,f,g);
+    y = lo_S_inv * rhs; // Solve for y
+    x = postprocess_schur_solution (lo_A_inv,lo_B,y,f);
+   @endcode
+ * Note that due to the construction of @c S_inv_approx and subsequently
+ * @c S_inv, there are a pair of nested iterative solvers which could
+ * collectivly consume a lot of resources.
+ * Therefore care should be taken in the the choices leading to the construction
+ * of the iterative inverse_operators.
+ * One would typically consider the use of a IterationNumberControl
+ * (or a similar mechanism) to limit the number of inner solver iterations.
+ * This controls the accuracy of the approximate inverse operation
+ * \f$ \tilde{S}^{-1} \f$ which acts only as the precondtioner for
+ * \f$ S^{-1} \f$.
+ * Furthermore, the preconditioner to \f$ \tilde{S}^{-1} \f$, which in this example is
+ * \f$ \text{prec}(D) \f$, should ideally be computationally inexpensive.
```

Comment:
I agree with @bangerth. In addition, I think that providing a test case like step-22 (but a very small mesh, say a 2x2 mesh) would also be very nice. I think you should be able to copy most parts of step-22 for that (I would suggest you use ILU instead of UMFPACK to avoid a dependency on umfpack for the test case).